### PR TITLE
MatchingDeclarationName. KtFilesSpec also remove .common.kt suffix from kotlin files

### PR DESCRIPTION
**Pull request by lexa-diky**
No changes for rule itself, but MatchingDeclarationName now will threat `.common.kt` suffix as known kotlin file part.

### CODE REVIEW

File: detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt [hunk 1]
```diff
@@ -10,16 +10,29 @@ import java.nio.file.Paths
...
+    KOTLIN_SUFFIX,
+    KOTLIN_SCRIPT_SUFFIX
+)
 
 val PsiFile.fileName: String
     get() = name.substringAfterLast(File.separatorChar)
 
+/**
+ * Removes kotlin specific file name suffixes, e.g. .kt.
+ * Note, will note remove other possible/knowb file suffixes like .java
```
Comment (BraisGabin) [inline detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt:26]:
```suggestion
 * Note, will not remove other possible/knowb file suffixes like .java
```

File: detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt [hunk 2]
```diff
@@ -10,16 +10,29 @@ import java.nio.file.Paths
 
 const val KOTLIN_SUFFIX = ".kt"
 const val KOTLIN_SCRIPT_SUFFIX = ".kts"
+const val KOTLIN_KMP_COMMON_SUFFIX = ".common.kt"
+
+val KOTLIN_FILE_SUFFIXES = arrayOf(
```
Comment (BraisGabin) [inline detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt:15]:
```suggestion
private val KOTLIN_FILE_SUFFIXES = arrayOf(
```

File: detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt [hunk 3]
```diff
@@ -10,16 +10,29 @@ import java.nio.file.Paths
 
 const val KOTLIN_SUFFIX = ".kt"
 const val KOTLIN_SCRIPT_SUFFIX = ".kts"
+const val KOTLIN_KMP_COMMON_SUFFIX = ".common.kt"
```
Comment (BraisGabin) [inline detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt:13]:
```suggestion
private const val KOTLIN_KMP_COMMON_SUFFIX = ".common.kt"
```

**Review by BraisGabin — COMMENTED** [review]
Looks good, thanks! I just left some minor things.

File: detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt (hunk 3 above)
Comment (lexa-diky) [reply detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt]: fixed

File: detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt (hunk 2 above)
Comment (lexa-diky) [reply detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt]: fixed

File: detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt (hunk 1 above)
Comment (lexa-diky) [reply detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt]: fixed, also knowb -> known

**Review by cortinico — APPROVED** [review]

**Review by BraisGabin — APPROVED** [review]

**Pull request merged**